paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala

```diff
@@ -69,10 +69,20 @@ trait PaimonPartitionManagement extends SupportsAtomicPartitionManagement {
       case fileStoreTable: FileStoreTable =>
         val partitions = toPaimonPartitions(rows).map(_.asInstanceOf[JMap[String, String]])
         val commit: FileStoreCommit = fileStoreTable.store.newCommit(UUID.randomUUID.toString)
+        var metastoreClient: MetastoreClient = null
+        val clientFactory = fileStoreTable.catalogEnvironment().metastoreClientFactory
         try {
           commit.dropPartitions(partitions.toSeq.asJava, BatchWriteBuilder.COMMIT_IDENTIFIER)
+          // sync to metastore with delete partitions
+          if (clientFactory != null) {
```

Comment:
should check `metastore.partitioned-table`.

Reply:
Yes，had added a check for metastore.partitioned-table and case. Thanks @JingsongLi